### PR TITLE
metric calc

### PR DESCRIPTION
**Pull request by yotamazriel**
- **Add resources request and limit to streaming handler**
- **Add resources request and limit to redis**
- **Add ARBAC authorization for engine to create deployments**
- **Update config map with metric runner requierment**

### CODE REVIEW

**Review by snir-shem-tov — APPROVED** [review]

File: charts/tensorleap/charts/engine/templates/streaming-handler-deployment.yaml [hunk 1]
```diff
@@ -25,6 +25,11 @@ spec:
     spec:
       containers:
         - image: {{ .Values.image }}:{{ .Values.image_tag }}
+          resources:
+            requests:
+              memory: 500Mi
+            limits:
+              memory: 1Gi
```
Comment (asafyehezkel) [inline charts/tensorleap/charts/engine/templates/streaming-handler-deployment.yaml:32]: we should be able to add these params to settings such as engine job template override

File: charts/tensorleap/charts/engine/templates/engine_sa.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: tensorleap
+  name: deployment-manager-role
+rules:
+- apiGroups: ["apps", ""]
+  resources: ["deployments"]
+  verbs: ["create", "delete", "get", "list", "watch", "update"]
```
Comment (asafyehezkel) [inline charts/tensorleap/charts/engine/templates/engine_sa.yaml:15]: I think some verbs are redundant here but it is meaningless

File: charts/tensorleap/charts/engine/templates/engine_sa.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+apiVersion: v1
```
Comment (asafyehezkel) [inline charts/tensorleap/charts/engine/templates/engine_sa.yaml:1]: don't forget to add it to `engine` repo as well

File: charts/tensorleap/charts/engine/templates/engine-cm.yaml [hunk 4]
```diff
@@ -24,5 +24,9 @@ data:
   CONTENT_BASE_URL: /session/
   UI_SETTINGS_SCHEMA: '{"N_DATASET_PROCESSES":{"type":"number","title":"Number of Dataset Processes","min":1,"description":"Number of processes to use for dataset processing (if empty, calculated automatically)"},"N_METRICS_PROCESSES":{"type":"number","title":"Number of Metrics Processes","min":1,"description":"Number of processes to use for metrics processing (if empty, calculated automatically)"},"N_VISUALIZERS_PROCESSES":{"type":"number","title":"Number of Visualizers Processes","min":1,"description":"Number of processes to use for visualizers processing (if empty, calculated automatically)"}}'
 
+  METRIC_CALCULATOR_IMAGE: {{ .Values.image_name }}
+  METRIC_CALCULATOR_IMAGE_TAG: {{ .Values.image_tag }}
+  METRIC_CALCULATOR_ENTRY_POINT: {{ .Values.entry_point }}
```
Comment (asafyehezkel) [inline charts/tensorleap/charts/engine/templates/engine-cm.yaml:29]: do we have more than one entry point?

**Review by asafyehezkel — APPROVED** [review]
LGTM
Added some comments

**Pull request merged**